File: pm/src/main/java/org/sqlbuilder/pm/objects/FnRoleAlias.java
package org.sqlbuilder.pm.objects;

import org.sqlbuilder.common.Resolvable;
import org.sqlbuilder.common.Utils;
import org.sqlbuilder2.ser.Triplet;

public class FnRoleAlias implements Resolvable<Triplet<String,String,String>, Triplet<Integer,Integer,Integer>>
{
	public String frame;

	public String fetype;

	public String lu;

	@Override
	public String dataRow()
	{
		return String.format("'%s',%s,%s",frame, Utils.nullableQuotedEscapedString(fetype), Utils.nullableQuotedEscapedString(lu));
	}

	@Override
	public Triplet<String,String, String> resolving()
	{
		return new Triplet<>(frame, fetype, lu);
	}
}

File: pm/src/main/java/org/sqlbuilder/pm/objects/FnRoleAlias.kt
package org.sqlbuilder.pm.objects

import org.sqlbuilder.common.Resolvable
import org.sqlbuilder.common.Utils.nullableQuotedEscapedString
import org.sqlbuilder.pm.PmFnRoleAndLuResolvable
import org.sqlbuilder.pm.PmFnRoleAndLuResolved

class FnRoleAlias : Resolvable<PmFnRoleAndLuResolvable, PmFnRoleAndLuResolved> {

    var frame: String? = null

    var fetype: String? = null

    var lu: String? = null

    override fun dataRow(): String {
        return "${nullableQuotedEscapedString(frame)},${nullableQuotedEscapedString(fetype)},${nullableQuotedEscapedString(lu)}"
    }

    override fun resolving(): PmFnRoleAndLuResolvable {
        return PmFnRoleAndLuResolvable(frame!!, fetype!!, lu!!)
    }
}
